components/script/dom/eventtarget.rs

```diff
@@ -401,12 +408,24 @@ impl EventTarget {
                     entries.push(EventListenerEntry {
                         phase: ListenerPhase::Bubbling,
                         listener: EventListenerType::Inline(listener),
+                        once: false
                     });
                 }
             },
         }
     }
 
+    pub fn remove_listener_if_once(&self, ty: &Atom, listener: Rc<EventListener>) {
```

Comment:
Make `listener` be a `&Rc<EventListener>` and you will be able to avoid a cloning operation in the caller.